curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-map-data-structure.english.md

```diff
@@ -70,7 +70,44 @@ var Map = function() {
 <section id='solution'>
 
 ```js
-// solution required
+var Map = function() {
+    this.collection = {};
+    // change code below this line
+
+    this.add = function(key,value) {
+      this.collection[key] = value;
+    }
+
+    this.remove = function(key) {
+      delete this.collection[key];
+    }
+
+    this.get = function(key) {
+      return this.collection[key];
+    }
+
+    this.has = function(key) {
+      if (this.collection.hasOwnProperty(key)) {
+        return true;
+      }
+      return false;
```

Comment:
This could also be simplified to `return this.collection.hasOwnProperty(key)`.

Reply:
@Manish-Giri It could, but the solution works and it is readable.  Are you ok with merging it as is?

Reply:
@RandellDawson Yup, it's good to merge. I was just making a note there.

Reply:
It cannot be unseen now, I was to busy adding spaces and semi-colons.